java/src/main/java/io/cucumber/gherkin/GherkinLine.java

```diff
@@ -70,35 +71,60 @@ boolean startsWith(String prefix) {
         return text.startsWith(prefix);
     }
 
+    public boolean startsWith(char prefix) {
+        return !empty && text.charAt(0) == prefix;
+    }
+
     String substringTrimmed(int beginIndex) {
-        return text.substring(beginIndex).trim();
+        // trim the beginning of the line (the end of line has already been trimmed in the constructor)
+        while ((beginIndex < textLength)) {
+            char c = text.charAt(beginIndex);
+            if (!(c <= ' ' || StringUtils.isWhiteSpaceExtended(c))) break;
```

Comment:
On reflection, between https://github.com/cucumber/gherkin/pull/442 catching most of the space like characters `< ' '` and the other characters being line noise, I would be happy breaking this.

Please do extract this as `leftTrim` to StringUtils`.